bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/utils/IdentifierUtils.java

```diff
@@ -30,7 +30,7 @@
  */
 public class IdentifierUtils {
 
-    private static final String UNICODE_REGEX = "\\\\u\\{([a-fA-F0-9]+)\\}";
+    private static final String UNICODE_REGEX = "(\\\\*)\\\\u\\{([a-fA-F0-9]+)\\}";
```

Comment:
Please check if we can move the leading back slash pattern after first back slash, I think it will reduce some back tracking. (I'm not 100% sure about this, just an idea)
```\\\\(\\\\*)u\\{([a-fA-F0-9]+)\\}```

Reply:
I see. yeah, we can move.